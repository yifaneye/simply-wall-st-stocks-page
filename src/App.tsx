import React from 'react';

interface Props {
  name?: string;
}
const App: React.FunctionComponent<Props> = ({ name }) => <h1>The {name}</h1>;
App.defaultProps = {
  name: 'Grid',
};
export { App };
