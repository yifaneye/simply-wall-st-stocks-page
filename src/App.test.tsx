import React from 'react';
import renderer from 'react-test-renderer';

import { App } from './App';

describe('The stock page', () => {
  // unit test
  it('is truthy', () => {
    expect(App).toBeTruthy();
  });

  // snapshot test
  it('renders the same as snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
