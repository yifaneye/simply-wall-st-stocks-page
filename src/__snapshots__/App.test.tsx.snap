// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The stock page renders the same as snapshot 1`] = `
<div
  className="sc-iqAclL kjoUJG"
>
  <div
    className="sc-jrsJWt biDlNd"
  >
    <div
      className=" css-2b097c-container"
      onKeyDown={[Function]}
    >
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className=" css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-g1d714-ValueContainer"
        >
          <div
            className=" css-1uccc91-singleValue"
          >
            Australia
          </div>
          <div
            className="css-b8ldur-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-kEqXSa iRqFgG"
  >
    <div
      className=" css-2b097c-container"
      onKeyDown={[Function]}
    >
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        className="css-1f43avz-a11yText-A11yText"
      />
      <div
        className=" css-yk16xz-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-g1d714-ValueContainer"
        >
          <div
            className=" css-1uccc91-singleValue"
          >
            Market Cap High to Low
          </div>
          <div
            className="css-b8ldur-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-3-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className=" css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-tj5bde-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  Loading ...
  <div
    className="sc-fujyAs iydsox"
  >
    <a
      className="sc-iCoGMd jNqhHG"
      disabled={true}
      href={null}
    >
      <svg
        fill="#fff"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M7.78127 11.29L14.8283 4.24303C15.2188 3.8525 15.852 3.8525 16.2425 4.24303C16.633 4.63355 16.633 5.26672 16.2425 5.65724L9.87891 12.0208L16.2425 18.3844C16.633 18.7749 16.633 19.4081 16.2425 19.7986C15.852 20.1891 15.2188 20.1891 14.8283 19.7986L7.78057 12.7509C7.7727 12.7435 7.76491 12.736 7.75723 12.7283C7.56873 12.5398 7.47122 12.2948 7.46468 12.0478C7.45748 11.7828 7.55498 11.5156 7.7572 11.3133C7.76513 11.3054 7.77315 11.2977 7.78127 11.29Z"
          fillRule="evenodd"
        />
      </svg>
    </a>
    <a
      className="sc-iCoGMd jNqhHG"
      disabled={true}
      href={null}
    >
      <svg
        fill="#fff"
        height="24"
        style={
          Object {
            "transform": "rotate(180deg)",
          }
        }
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M7.78127 11.29L14.8283 4.24303C15.2188 3.8525 15.852 3.8525 16.2425 4.24303C16.633 4.63355 16.633 5.26672 16.2425 5.65724L9.87891 12.0208L16.2425 18.3844C16.633 18.7749 16.633 19.4081 16.2425 19.7986C15.852 20.1891 15.2188 20.1891 14.8283 19.7986L7.78057 12.7509C7.7727 12.7435 7.76491 12.736 7.75723 12.7283C7.56873 12.5398 7.47122 12.2948 7.46468 12.0478C7.45748 11.7828 7.55498 11.5156 7.7572 11.3133C7.76513 11.3054 7.77315 11.2977 7.78127 11.29Z"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </div>
</div>
`;
